### ISSUE TITLE
Can't start Umami on Docker since 2.11.0 

### ISSUE THREAD

**Issue by eszpee** [umami-software/umami#2651]
### Describe the Bug

Umami keeps on trying to restart, the pasted log output is being repeated in the container log. 

Relevant docker compose parts (though it hasn't changed since before the upgrade)

```
  umami:
    env_file: .env
    image: ghcr.io/umami-software/umami:postgresql-latest
    ports:
      - "3000:3000"
    environment:
      DATABASE_URL: $UMAMI_DATABASE_URL
      DATABASE_TYPE: postgresql
      APP_SECRET: $UMAMI_APP_SECRET
    depends_on:
      umami-db:
        condition: service_healthy
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "curl http://localhost:3000/api/heartbeat"]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - frontend
      - backend

  umami-db:
    env_file: .env
    image: postgres:15-alpine
    environment:
      POSTGRES_DB: umami
      POSTGRES_USER: $UMAMI_DB_USER
      POSTGRES_PASSWORD: $UMAMI_DB_PASSWORD
    volumes:
      - umami-db-data:/var/lib/postgresql/data
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U $${POSTGRES_USER} -d $${POSTGRES_DB}"]
      interval: 5s
      timeout: 5s
      retries: 5
    networks:
      - backend
```

Any help appreciated, I might be doing something wrong. Thanks!

### Database

PostgreSQL

### Relevant log output

```shell
yarn run v1.22.19
$ npm-run-all check-db update-tracker start-server
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✓ Database connection successful.
✓ Database version check successful.
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "umami", schema "public" at "umami-db:5432"
5 migrations found in prisma/migrations
No pending migrations to apply.
✓ Database is up to date.
$ node scripts/update-tracker.js
$ node server.js
 ⨯ Failed to start server
Error: listen EADDRNOTAVAIL: address not available 192.168.100.101:3000
    at Server.setupListenHandle [as _listen2] (node:net:1800:21)
    at listenInCluster (node:net:1865:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:2014:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:8) {
  code: 'EADDRNOTAVAIL',
  errno: -99,
  syscall: 'listen',
  address: '192.168.100.101',

  port: 3000
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "start-server" exited with 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
```

### Which Umami version are you using? (if relevant)

2.11.0

### How are you deploying your application? (if relevant)

Docker compose

**Comment by AkashRajpurohit** [comment]
Seems like the port 3000 is not available, is something else running on that port?
You can use netstat to check it -> `netstat -tulanp`

**Comment by eszpee** [comment]
Thanks for the quick reply! Nothing ran on 3000, but just to be safe, I stopped all other containers on this VM, and Umami is still not starting. 

There’s something weird, it seems to try binding to 192.168.100.101. That’s not the IP of the container or the host neither. Where is Umami getting this address from? Is it maybe stored in the db somehow? I did a database export/import from another instance a few weeks ago, it was working perfectly until now. Maybe the db change with the version upgrade uncovered something related?

Thanks again for the help, let me know if there’s a better platform for this, I don’t want to pollute these Issues with something that doesn’t seem like a bug.

**Comment by eszpee** [comment]
```
yarn run v1.22.19
$ npm-run-all check-db update-tracker start-server
$ node scripts/check-db.js
✓ DATABASE_URL is defined.
✓ Database connection successful.
✓ Database version check successful.
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "umami", schema "public" at "umami-db:5432"
5 migrations found in prisma/migrations
No pending migrations to apply.
✓ Database is up to date.
$ node scripts/update-tracker.js
$ node server.js
 ⨯ Failed to start server
Error: listen EADDRNOTAVAIL: address not available 192.168.100.106:3000
    at Server.setupListenHandle [as _listen2] (node:net:1800:21)
    at listenInCluster (node:net:1865:12)
    at GetAddrInfoReqWrap.doListen [as callback] (node:net:2014:7)
    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:8) {
  code: 'EADDRNOTAVAIL',
  errno: -99,
  syscall: 'listen',
  address: '192.168.100.106',

  port: 3000
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "start-server" exited with 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
```

192.168.100.106 is the correct IP and port 3000 is not used:

```
root@selfhost:/home/eszpee/docker# docker start docker_umami_1 && docker exec docker_umami_1 netstat -tulanp
docker_umami_1
Active Internet connections (servers and established)
Proto Recv-Q Send-Q Local Address           Foreign Address         State       PID/Program name
tcp        0      0 127.0.0.11:43375        0.0.0.0:*               LISTEN      -
udp        0      0 127.0.0.11:49926        0.0.0.0:*                           -
```

**Comment by eszpee** [comment]
Nevermind, it was my fault, closing this with the solution: 

```
  umami:
    env_file: .env

```    
This surprisingly (for me) caused problems, commenting out the env_file line solved everything.
